playbook_ui/lib/src/playbook_gallery.dart

```diff
@@ -155,14 +159,26 @@ class _PlaybookGalleryState extends State<PlaybookGallery> {
   @override
   void didUpdateWidget(covariant PlaybookGallery oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (widget.searchTextController != oldWidget.searchTextController) {
+      oldWidget.searchTextController?.removeListener(_searchTextListener);
+      widget.searchTextController?.addListener(_searchTextListener);
+
+      if (widget.searchTextController == null) {
+        _defaultSearchTextController.clear();
```

Comment:
Does it need to clear here?

Reply:
Yes, we don't need to clear it.
But clearing it behaves more naturally, IMO.
Considering the following case:


- `widget.controller = null`, `state._defaultController != null`, `searchText == ''`
- Input something, then `state._defaultController.text == 'A'`, `searchText == 'A'`
- Set a non-null controller with `widget.controller.text == 'B'`, `searchText == 'B'`
- Set a null controller, so `widget.controller == null`
  - If we clear the _defaultController, then `searchText == ''`
  - If we do nothing, then `searchText == 'A'` again. Maybe a little bit weird?

(`searchText` means the search word we see in the UI.)

What do you think of it?

Reply:
>Isn't searchText == 'B' natural?
Inherit from widget.controller.text to state._defaultController.text.

Sounds good.

Reply:
@KyoheiG3 

Fixed